### ISSUE TITLE
components global:true results in: At least one <template> or <script> is required in a single file component

### ISSUE THREAD

**Issue by LucaMargadant** [storybook-vue/storybook-nuxt#4]
Thank you a lot for this package. It works very well and i'm super happy that we now have something for nuxt to work with storybook! 👍 

I noticed that if you have set global components in nuxt.config to true like this for example, `components: { global: true, dirs: ['~/components/global']},`, for every globally registered component there will be rendered an error `At least one <template> or <script> is required in a single file component`. 

If you click away the error the storybook and all stories are working as expected until you click on another story.

It's probably a low priority one, because global components are not recommended anyways, but in some circumstances global components can make sense so i just wanted to drop that one here.

**Comment by g1ronn1mo** [comment]
How did you install the package? I cant get it running :(

**Comment by chakAs3** [comment]
> Thank you a lot for this package. It works very well and I'm super happy that we now have something for Nuxt to work with storybook! 👍

Welcome, I'm really glad to hear that, I was aware of the importance of  Nuxt especially for scalable enterprise applications, I put it last in the Q2 roadmap because I had to rewrite the vue3 renderer in reactive mode first for v7,  there are other awesome features for Vue developers, Storybook will have a Vue taste and not just a second choice.  i am happy to put Vue as a first-class citizen in Storybook, I would love to get more feedback. from the community then I can improve Storybook for Vue / Nuxt

**Comment by vloss3** [comment]
due to this error, the vueApp is not being mounted(undefined) and composables fail:
```
runtime.js:4 Error: [nuxt] A composable that requires access to the Nuxt instance was called outside of a plugin, Nuxt hook, Nuxt middleware, or Vue setup function. This is probably not a Nuxt bug. Find out more at ...
```

**Comment by oliver-lehner** [comment]
I run into the same error but I haven't set up global components explicitly. I'm using [`@storyblok/nuxt`](https://github.com/storyblok/storyblok-nuxt) though and removing the module from my nuxt config makes the error go away.

**Comment by rezkileap** [comment]
Any updates on this issue?

**Comment by g1ronn1mo** [comment]
https://github.com/storybook-vue/storybook-nuxt

Check out this Repo. It still have bugs, but works for most cases. It is important to notice, that it only runs on Linux, and NOT on Windows.

**Comment by chakAs3** [comment]
@rezkileap  i can't reproduce your bug,. Here NuxtWelome component working fine as well as the Entire Vuetify components which are all global, please explain show there are global
![image](https://github.com/storybook-vue/storybook-nuxt/assets/711292/0be19df2-0ffd-4dbc-863c-edaf1b964d82)

**Comment by rezkileap** [comment]
I've boostrap the Nuxt authoring module repo, then in the playground folder, set up the Storybook, I used 'npx storybook-nuxt init' and opted for pnpm. Now, everything was working until I tried to import a component that required Nuxt Image or one that depended on another component. I've linked a simple repo to replicate this issue. I hope this make sense 
https://github.com/rezkileap/nuxt-module-authoring-kit-with-sb7

**Comment by rezkileap** [comment]
I've boostrap the Nuxt authoring module repo, then in the playground folder, set up the Storybook, I used 'npx storybook-nuxt init' and opted for pnpm. Now, everything was working until I tried to import a component that required Nuxt Image or one that depended on another component. Ive attached a simple repo to replicate this issue. 

https://github.com/rezkileap/nuxt-module-authoring-kit-with-sb7

**Comment by vloss3** [comment]
@chakAs3 sadly it still doesn't work, in @rezkileap example:
https://stackblitz.com/~/github.com/rezkileap/nuxt-module-authoring-kit-with-sb7
run:
`cd playground && npm install && npm run build-storybook`

```
ERROR  [vite:vue] At least one <template> or <script> is required in a single file component.                            19:04:38
file: /home/rezkileap/nuxt-module-authoring-kit-with-sb7/src/runtime/components/testComp.vue?nuxt_component=async

=> Failed to build the preview                                                                                            19:04:38
SyntaxError: At least one <template> or <script> is required in a single file component.                                  19:04:38
    at Object.parse$2 [as parse] (file://./node_modules/@vue/compiler-sfc/dist/compiler-sfc.cjs.js:1904:7)
    at createDescriptor (file://./node_modules/@vitejs/plugin-vue/dist/index.mjs:90:43)
    at transformMain (file://./node_modules/@vitejs/plugin-vue/dist/index.mjs:2302:34)
    at Object.transform (file://./node_modules/@vitejs/plugin-vue/dist/index.mjs:2834:16)
    at eval (file://./node_modules/rollup/dist/es/shared/node-entry.js:25569:40)
                                                                                                                          19:04:38
? Would you like to help improve Storybook by sending anonymous crash reports? › (Y/n)                                    19:04:38
```

In the example above the components are added via module:
```js
await addComponentsDir({
  path: resolve('./runtime/components/'),
  global: true,
  pathPrefix: true,
  prefix: 'Mds',
 })
```

but it also doesn't work if you add components directly to `./components/global/component.vue` which makes them global

**Comment by cyruscollier** [comment]
I've having the same issue. Our Nuxt 3 application requires some global components due to a need to dynamically use `resolveComponent()` with a component name as a variable rather than a literal string.

I did some digging on what might be happening when Nuxt is loaded through Storybook using this add-on. When Nuxt is started normally, the Vue compiler receives a component's source, which is just the component file's raw content as a string, e.g. `<template><button>Test</button></template><script></script>`. However, when Storybook loads Nuxt, that source string for global components appears to be an async component export, e.g. 
`import { defineAsyncComponent } from "vue"\n
export default defineAsyncComponent(() => import("/Users/ccollier/Develop/test-project/components/global/GlobalTest.vue").then(r => r.default))`

It looks like there's some missing step where Storybook and/or Vite needs to actually evaluate that dynamic export so that the source is the actual file content. Perhaps Storybook isn't expecting these global components to be auto-imported at startup and isn't handling it correctly? I'm not sure how to fix this (yet), but hopefully this might point the maintainers in the right direction.

**Comment by rpowersstl** [comment]
@cyruscollier  I am having the same problem 

' ERROR  Internal server error: At least one <template> or <script> is required in a single file component.
  Plugin: vite:vue'

Did you find a fix for this?

**Comment by usrrname** [comment]
I can confirm I've come across the same problem in nuxt 3.6.5 with vue 3 components using composition api

**Comment by markus-gx** [comment]
Still encounter the same error

```
 ERROR  Internal server error: At least one <template> or <script> is required in a single file component.
  Plugin: vite:vue
```

It seems it happens due to global components which OP already noticed / explained. But sadly there are some moments you need them :/ 
Any solutions at all?

**Comment by HPieters** [comment]
Not sure if it's helpful but I am running into the same issue and can confirm previous comments. 

As mentioned before, and also for me, this is caused by a custom module that is using `addComponentsDir` with `global: true`. If I disable my custom module everything works great again.

**Comment by Ibochkarev** [comment]
@chakAs3 Good afternoon

Tell me if there is any news on correcting this problem indicated in the essay and in https://github.com/storybookjs/storybook/issues/21196.

For many months, this has been blocking the launch of a storybook in a large project.

If there is an opportunity to somehow prioritize this problem, we would all be grateful!

**Comment by chakAs3** [comment]
Hi @Ibochkarev sorry for the inconvenient, i will prioritize this problem doing my best to unblock your launch.

**Comment by lisaschumann** [comment]
I'm running into the same issue when using nuxt-icons :/ `At least one <template> or <script> is required in a single file component.` Is there an ETA on a fix?

**Comment by markus-gx** [comment]
> I'm running into the same issue when using nuxt-icons :/ `At least one <template> or <script> is required in a single file component.` Is there an ETA on a fix?

I personally switched to [Histoire](https://histoire.dev), which works pretty well. Got everything setup within 30mins. Great solution until this is fixed on Storybook' Side.

**Comment by chakAs3** [comment]
Sorry I'm busy with other issues, I will check this as soon as possible. It is mainly related to vite plugin versions

**Comment by Ibochkarev** [comment]
@chakAs3 Good afternoon!
Is there any news on resolving this blocker?

**Comment by chakAs3** [comment]
@Ibochkarev  did you use the latest version as it is working fine . let me know if the issue still occurs

**Comment by rpauls** [comment]
Quickly checked `storybook-nuxt` by throwing it via `npx storybook-nuxt init` into an existing larger Nuxt project (using `nuxt@3.11.1` and `vue@3.4.21`.

The error appears to be persisting:

```sh
ERROR  Internal server error: At least one <template> or <script> is required in a single file component.
  Plugin: vite:vue
  File: /path/to/project/nuxt/modules/subscription/components/ProductSubscriptionSelection.vue
      at Object.parse$2 [as parse] (/path/to/project/nuxt/node_modules/@vue/compiler-sfc/dist/compiler-sfc.cjs.js:1921:7)
      at createDescriptor (/path/to/project/nuxt/node_modules/@storybook-vue/nuxt/dist/preset.cjs:87:43)
      at transformMain (/path/to/project/nuxt/node_modules/@storybook-vue/nuxt/dist/preset.cjs:2380:34)
      at TransformContext.transform (/path/to/project/nuxt/node_modules/@storybook-vue/nuxt/dist/preset.cjs:2923:16)
      at Object.transform (file:///path/to/project/nuxt/node_modules/vite/dist/node/chunks/dep-whKeNLxG.js:51172:62)
      at async loadAndTransform (file:///path/to/project/nuxt/node_modules/vite/dist/node/chunks/dep-whKeNLxG.js:53923:29)
```

**Comment by gavmck** [comment]
Just updated to `@storybook-vue/nuxt` here as part of a nuxt 3 upgrade. We have a global component folder for storyblok components (that are not even documented in storybook) and this error blows up storybook.

**Comment by Ibochkarev** [comment]
@tobiasdiez @chakAs3  Hello!

In https://github.com/nuxt-modules/storybook/releases/tag/v8.1.1 the error continues to block the use of Storybook

```bash
 ERROR  Internal server error: At least one <template> or <script> is required in a single file component.                                                                        2:26:19 PM
  Plugin: vite:vue
  File: /Users/ibochkarev/Projects/v3/src/widgets/templates/w-feed-template/w-video-section-template.vue
      at Object.parse$2 [as parse] (/Users/ibochkarev/Projects/v3/node_modules/.pnpm/@vue+compiler-sfc@3.4.21/node_modules/@vue/compiler-sfc/dist/compiler-sfc.cjs.js:1921:7)
      at createDescriptor (/Users/ibochkarev/Projects/v3/node_modules/.pnpm/@storybook-vue+nuxt@0.2.6_@types+node@18.19.31_encoding@0.1.13_eslint@8.57.0_meow@9.0.0_nuxt@_hv7zc5tdoxa7qtmqb6wsfxacpq/node_modules/@storybook-vue/nuxt/dist/preset.cjs:87:43)
      at transformMain (/Users/ibochkarev/Projects/v3/node_modules/.pnpm/@storybook-vue+nuxt@0.2.6_@types+node@18.19.31_encoding@0.1.13_eslint@8.57.0_meow@9.0.0_nuxt@_hv7zc5tdoxa7qtmqb6wsfxacpq/node_modules/@storybook-vue/nuxt/dist/preset.cjs:2380:34)
      at TransformContext.transform (/Users/ibochkarev/Projects/v3/node_modules/.pnpm/@storybook-vue+nuxt@0.2.6_@types+node@18.19.31_encoding@0.1.13_eslint@8.57.0_meow@9.0.0_nuxt@_hv7zc5tdoxa7qtmqb6wsfxacpq/node_modules/@storybook-vue/nuxt/dist/preset.cjs:2923:16)
      at Object.transform (file:///Users/ibochkarev/Projects/v3/node_modules/.pnpm/vite@5.2.8_@types+node@18.19.31_sass@1.76.0_terser@5.30.3/node_modules/vite/dist/node/chunks/dep-whKeNLxG.js:51172:62)
      at async loadAndTransform (file:///Users/ibochkarev/Projects/v3/node_modules/.pnpm/vite@5.2.8_@types+node@18.19.31_sass@1.76.0_terser@5.30.3/node_modules/vite/dist/node/chunks/dep-whKeNLxG.js:53923:29)
```

**Comment by alex-cailler** [comment]
@tobiasdiez @chakAs3  Hello ! 

I Have the same problem of @Ibochkarev, with project installed yesterday.

```
ERROR  Internal server error: At least one <template> or <script> is required in a single file component.
  Plugin: vite:vue
  File: /Users/cailler/Projects/molink/node_modules/nuxt-icon/dist/runtime/IconCSS.vue
      at Object.parse$2 [as parse] (/Users/cailler/Projects/molink/node_modules/@vue/compiler-sfc/dist/compiler-sfc.cjs.js:1923:7)
      at createDescriptor (/Users/cailler/Projects/molink/node_modules/@storybook-vue/nuxt/dist/preset.cjs:87:43)
      at transformMain (/Users/cailler/Projects/molink/node_modules/@storybook-vue/nuxt/dist/preset.cjs:2380:34)
      at TransformContext.transform (/Users/cailler/Projects/molink/node_modules/@storybook-vue/nuxt/dist/preset.cjs:2923:16)
      at Object.transform (file:///Users/cailler/Projects/molink/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:52318:62)
      at async loadAndTransform (file:///Users/cailler/Projects/molink/node_modules/vite/dist/node/chunks/dep-DEPSZ3SS.js:55073:29)
```

**Comment by Archetipo95** [comment]
Still present in latest

**Comment by andreaswolf** [comment]
I also see this with latest nightly (8.4.2-29212946.04b4f98) and Storybook 9.0.18, and it's effectively preventing us from using Storybook with Nuxt components (since we need to register some components globally for our TYPO3 Integration with https://github.com/TYPO3-Headless/nuxt-typo3).

**Comment by andreaswolf** [comment]
IIRC I could fix the issue by removing the leftover vite.config.ts file.
